lib/manageiq/session_store.rb

```diff
@@ -0,0 +1,88 @@
+module ManageIQ
+  module SessionStore
+    def self.store
+      @store
```

Comment:
I was thinking this bit may not be needed because I was expecting a way to access the session store instance directly from Rails (via somethiing like `Rails.applications.???`), but I just cannot find it, so this may be needed.   Will keep digging.

Reply:
Yeah, I tried looking as well (a lot of time spent on this was doing just that), but I came up empty.  Everything I found was around using this as part of a `middleware`, which assumes being part of he `rack` app interface and using the `.call(app, env)` interface.

This circumvents that, but the `@store` interface was required to do that.

Reply:
Here's what I got:

```ruby
def detect_middleware_instance(mw, klass)
  mw.is_a?(klass) ? mw : detect_middleware_instance(mw.instance_variable_get(:@app), klass)
end

detect_middleware_instance(Rails.application, ActionDispatch::Session::SessionObject)
# => => #<ActionDispatch::Session::MemCacheStore:0x00007fe51f770f90 ...>
```

This is admittedly horribly gross...so much better to just have something like the .store accessor.

Even so, what is being returned by the above is not the same instance as what is returned by ManageIQ::SessionStore.store (same config options, so maybe it doesn't matter)

Reply:
Ended up going this route to set `@store`, though I don't know that it really matters that we did instead of creating a new instance with the same info.  Since there is a good chance this could be accessed from a different server that wouldn't have the same instance anyway, so any `mutex` locking or what have you wouldn't matter anyway (which is the only advantage sharing an instance might provide.

That said, the only reason I still went this route was that after all of the refactoring that happened as a result of the review process, I didn't have to add an extra method to each of the adapters (similar to `.class_to_patch`).  So it was one slightly ugly method instead of multiple tiny, but mostly useless, methods.